default_validator/src/lib.rs

```diff
@@ -7,9 +7,128 @@ extern crate alloc;
 #[cfg(not(test))]
 extern crate panic_halt;
 
-pub mod isi;
-
 use iroha_validator::{pass_conditions, prelude::*};
+use isi::{
+    account::{
+        validate_burn_account_public_key, validate_mint_account_public_key,
+        validate_mint_account_signature_check_condition, validate_remove_account_key_value,
+        validate_set_account_key_value, validate_unregister_account,
+    },
+    asset::{
+        validate_burn_asset, validate_mint_asset, validate_register_asset,
+        validate_remove_asset_key_value, validate_set_asset_key_value, validate_transfer_asset,
+        validate_unregister_asset,
+    },
+    asset_definition::{
+        validate_remove_asset_definition_key_value, validate_set_asset_definition_key_value,
+        validate_transfer_asset_definition, validate_unregister_asset_definition,
+    },
+    domain::{
+        validate_remove_domain_key_value, validate_set_domain_key_value, validate_unregister_domain,
+    },
+    parameter::{validate_new_parameter, validate_set_parameter},
+    peer::validate_unregister_peer,
+    permission::{
+        validate_grant_account_permission, validate_register_permission_token,
+        validate_revoke_account_permission,
+    },
+    role::{validate_grant_account_role, validate_revoke_account_role, validate_unregister_role},
+    trigger::{
+        validate_execute_trigger, validate_mint_trigger_repetitions, validate_unregister_trigger,
+    },
+    validator::validate_upgrade_validator,
+};
+
+mod isi;
+
+/// Default validator
+///
+/// You can put state in your validator and use that state
+struct Validator;
+
+macro_rules! custom_impls {
+    ( $($validator:ident($operation:ty)),+ $(,)? ) => { $(
+        fn $validator(&mut self, authority: &AccountId, operation: &$operation) -> Verdict {
+            $validator(self, authority, operation)
+        } )+
+    }
+}
+
+impl Validate for Validator {
+    custom_impls! {
```

Comment:
Imo we should have minimal validator which is allow all and user will build on top of that.
In such case it will be possible to analyze behavior more easily because everything in one place.

Reply:
From my point of view this complicates the current design, because for a little modification inside general well-tests `default_validator` user will have to copy-pase everything from it. So instead of being repsonsible for just one method override user will be responsible for all from `default_validator` too.

What about making `DefaultValidator` trait which will work at the same way as `Validator` in the `iroha_validator`?
So that my case and your case would work.

Something like this:

```rust
pub trait DefaultValidator {
    // Same functions as in `Validator`
}

impl<T: DefaultValidator> Validator for T {
    // Fallback to impls from `DefaultValidator`
}
```

Reply:
I'l take your suggestion @Arjentix

Reply:
now we have the best of both worlds